### ISSUE TITLE
Magic Mirror does not launch when module enabled

### ISSUE THREAD

**Issue by dani4ko03** [btastic/MMM-moonraker#4]
When the module is installed and the configurations are added to the config file the magic mirror stops opening. When I refresh it it just shows "Loading..." and then when i restart it it just doesn't open again. Everything goes back to normal when I delete it from the config file. I already ran a config check and my syntax is fine. I also added the ip of the pi of the mirror to that of the printer and still does the same issue. Not sure where to look and what else to do.

**Comment by btastic** [comment]
Hey, can you check your logs? `pm2 logs` and then `pm2 start MagicMirror`

That way we can see where the application will crash.. not sure how else to figure this one out.

**Comment by dani4ko03** [comment]
so it seems to get online fine when restarted
![image](https://github.com/btastic/MMM-moonraker/assets/132539573/697285c8-ea61-41e7-93ab-723d5a11ae0c)

but in the logs it seems it never finish loading all the modules and the mirror does not show up on my monitor it just sits at the home screen of the pi. 
these are the logs
![image](https://github.com/btastic/MMM-moonraker/assets/132539573/398e0bdb-a756-48ea-a81f-4598fc0c2e50)

my best guess is there is a conflict with another module but If im honest i dont fully understand the error logs.

**Comment by btastic** [comment]
So if you want to know what is happening, you can approach the stack trace here.

You usually read it from bottom to top. And you're basically looking for something you "know" or can access. In this case it would be the `... MMM-moonraker/node_helper.js:10:15`.

The interesting part about this is the 10 and 15. 10 means Line 10, 15 means character 15.

But in short: the fetch import is having issues, since in a newer version of MagicMirror, this is no longer included.

I tried fixing it, but then I was not able to make it work on older versions. Not sure how to proceed here.

**Comment by dani4ko03** [comment]
Isnt it fair to assume most mirror are getting updated regularly?

**Comment by btastic** [comment]
Yes, I'd say so. However, the update is not working for me. The MagicMirror main software was only crashing for me. I will have another look this week.

**Comment by dani4ko03** [comment]
let me try it for you if you want

**Comment by dani4ko03** [comment]
Im at my mirror right now. I can give it a try and let you know right away.

**Comment by dani4ko03** [comment]
![image](https://github.com/btastic/MMM-moonraker/assets/132539573/f918ff58-47a4-45b3-86c3-b39f3ab554fb)
looks like i made progress its not crashing any more but my printer is definitely online so im not sure

**Comment by btastic** [comment]
I went to bed last night. I can figure it out. I just need some more time.

**Comment by OlafdeViesel** [comment]
Hi guys. I found a solution.
Switch to "module/MMM-moonraker" and install "node-fetch" Package with `npm install node-fetch@2 --force`. Then open node_helper.js and change 
`const fetch = require("fetch");`
to
`const fetch = require("node-fetch");`

This is not the best solution but it works.

**Comment by dani4ko03** [comment]
Unfortunately, I'm doing something wrong because im still getting the Moonraker not available. Printer offline? message

**Comment by OlafdeViesel** [comment]
> Unfortunately, I'm doing something wrong because im still getting the Moonraker not available. Printer offline? message

You use the correct ip of the printer?

Show me your logs

**Comment by normig62** [comment]
@OlafdeViesel, bei mir geht es jetzt auch super, Danke :-)

**Comment by ijoshea** [comment]
> Hi guys. I found a solution. Switch to "module/MMM-moonraker" and install "node-fetch" Package with `npm install node-fetch@2 --force`. Then open node_helper.js and change `const fetch = require("fetch");` to `const fetch = require("node-fetch");`
> 
> This is not the best solution but it works.

Thanks! This still worked for me.

**Comment by TAGinside** [comment]
> Hi guys. I found a solution. Switch to "module/MMM-moonraker" and install "node-fetch" Package with `npm install node-fetch@2 --force`. Then open node_helper.js and change `const fetch = require("fetch");` to `const fetch = require("node-fetch");`
> 
> This is not the best solution but it works.

Worked nice for me thanks.
For the IP dont make the mistake to put the slash at end of IP address.

example : 
http://192.168.1.5/ = Wrong
http://192.168.1.5 = OK
